python/cuspatial/cuspatial/core/interpolate.py

```diff
@@ -0,0 +1,182 @@
+# Copyright (c) 2020, NVIDIA CORPORATION.
+
+import numpy as np
+
+from cudf import DataFrame, Series
+
+from cuspatial._lib.interpolate import (
+    cubicspline_coefficients,
+    cubicspline_interpolate,
+)
+
+
+def _cubic_spline_coefficients(x, y, ids, prefix_sums):
+    x_c = x._column
+    y_c = y._column
+    ids_c = ids._column
+    prefix_c = prefix_sums._column
+    result_table = cubicspline_coefficients(x_c, y_c, ids_c, prefix_c)
+    result = DataFrame._from_table(result_table)
+    return result
+
+
+def _cubic_spline_fit(points, points_ids, prefixes, original_t, c):
+    points_c = points._column
+    points_ids_c = points_ids._column
+    prefixes_c = prefixes._column
+    original_t_c = original_t._column
+    result_column = cubicspline_interpolate(
+        points_c, points_ids_c, prefixes_c, original_t_c, c
+    )
+    return result_column
+
+
+class CubicSpline:
+    """
+    Fits each column of the input Series `y` to a hermetic cubic spline.
+
+    Parameters
+    ----------
+    t : cudf.Series
+        time sample values. Must be monotonically increasing.
+    y : cudf.Series
+        columns to have curves fit to according to x
+    ids (Optional) : cudf.Series
+        ids of each spline
+    size (Optional) : cudf.Series
+        fixed size of each spline
+    prefixes (Optional) : cudf.Series
+        alternative to `size`, allows splines of varying
+        length. Not yet fully supported.
+
+    Returns
+    -------
+    CubicSpline object `o`. `o.c` contains the coefficients that can be
+    used to compute new points along the spline fitting the original `t`
+    data. `o(n)` interpolates the spline coordinates along new input
+    values `n`.
```

Comment:
Parameters and Returns should be in the `__init__` docstring

Reply:
Should I have any docs in the class, or put all docs in `__init__`?